kyaml/yaml/fns.go

```diff
@@ -259,7 +259,7 @@ func (f FieldMatcher) Filter(rn *RNode) (*RNode, error) {
 
 	// never match nil or null fields
 	if IsMissingOrNull(rn) {
-		return nil, nil
+		return nil, errors.Errorf("field has null value")
```

Comment:
Is this always an error that we can't recover from? It seems like users might want to check for this error and then just handle it appropriately.